### PR TITLE
handle only one touch and don't fail if another touch has ended

### PR DESCRIPTION
**Pull request by pankkor**
Is needed to handle scroll view gestures properly without failing drag gesture and cancelling drag operation

### CODE REVIEW

**Comment by frenetisch-applaudierend** [comment]
Hi pankkor

Thank you for your contribution. I'm not sure I understand what exactly this fixes. Can you give me an example of what you tried to do and it fails currently?

**Comment by pankkor** [comment]
Hi. If you drag from scroll view with one finger and want to interact with scroll view with another finger second finger tap cancels drag gesture because gesture recogniser doesn't track touches. Does it make any sense?

**Comment by pankkor** [comment]
this is not exactly correct, I'll make amendments soon

**Comment by frenetisch-applaudierend** [comment]
Thanks for the explanation, now I understand. That makes sense 👍

File: iOS Library/Sources/DNDLongPressDragRecognizer.m [hunk 1]
```diff
@@ -53,41 +52,49 @@ - (void)checkLongPress:(NSTimer *)timer {
...
 - (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event {
-    if (self.state == UIGestureRecognizerStateBegan || self.state == UIGestureRecognizerStateChanged) {
-        self.state = UIGestureRecognizerStateChanged;
-    } else if (self.state == UIGestureRecognizerStatePossible) {
-        CGPoint currentPoint = [self.trackedTouch locationInView:self.view];
-        CGPoint vector = CGPointMake(currentPoint.x - self.startPoint.x, currentPoint.y - self.startPoint.y);
-        CGFloat distance = (CGFloat)hypot(vector.x, vector.y);
-        
-        if (distance > self.allowableMovement) {
+    if ([touches containsObject:self.trackedTouch]) {
```
Comment (frenetisch-applaudierend) [inline iOS Library/Sources/DNDLongPressDragRecognizer.m]: If you're rewriting the code anyway, could you please change those big ifs into early returns?

**Comment by pankkor** [comment]
@frenetisch-applaudierend I've refactored DNDLongPressDragRecognizer a bit.
In summary: if multiple gesture recognizers interfere ```touchesEnded:``` or ````touchesCanelled:``` can be omitted for one of them. 
One shouldn't rely on those methods to reset/cleanup your gesture recognizer state. Instead one should override -reset method for you cleanup logic which is called even though ```touchesEnded:``` are ```touchesCanelled:``` not.

[reset documentation](https://developer.apple.com/reference/uikit/uigesturerecognizer/1620004-reset)

**Comment by lmirosevic** [comment]
Any chance of getting this merged?

**Comment by frenetisch-applaudierend** [comment]
Whoops, I'm so sorry this completely went under my radar. Give me a few days to wrap my head around it again

**Comment by frenetisch-applaudierend** [comment]
1.4.0 is now published on CocoaPods. Thanks again @pankkor

**Comment by lmirosevic** [comment]
Thanks to you both!

**Pull request merged**